src/Juvix/FrontendDesugar/RemoveDo/Transform.hs

```diff
@@ -0,0 +1,285 @@
+module Juvix.FrontendDesugar.RemoveDo.Transform where
+
+import qualified Data.List.NonEmpty as NonEmpty
+import qualified Juvix.FrontendDesugar.RemoveDo.Types as New
+import qualified Juvix.FrontendDesugar.RemovePunned.Types as Old
+import Juvix.Library
+
+-- The actual transform we are doing
+
+transformDo :: Old.Do -> New.Expression
+transformDo (Old.Do'' body) = foldr f (transformLast last) body
```

Comment:
... we need to also solve the problem of lifting application over an error monad

Reply:
... and a state monad

Reply:
maybe we should use `MonadIdentity` ( =/ )